client/my-sites/checkout/index.js

```diff
@@ -121,7 +121,7 @@ export default function () {
 	);
 
 	page(
-		`/checkout/marketplace/:productSlug`,
+		`/checkout/:productSlug/:intentId`,
```

Comment:
To ensure that this change applies only to the marketplace and passport, and to avoid unintended side effects, I suggest registering a new route with the `passport` hardcoded to manage the routing correctly.
```suggestion
        page(
                `/checkout/passport/:intentId`
                setLocaleMiddleware(),
		        noSite,
		        checkoutMarketplaceSiteless,
		        makeLayout,
		        clientRender
        )
        
        
		page(
		         `/checkout/marketplace/:intentId`,
```